wasm2c/wasm-rt-impl-mem.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2018 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This file is not meant to be used directly, and is only used by including it
+// in wasm-rt-impl.c. This file may be included multiple times to implement
+// memory operations on different memory tyeps --- wasm_rt_memory_t and
+// wasm_rt_memory_shared_t depending on the value of WASM_RT_SHARED_MEM_OPS.
+
+#include <wasm-rt-impl.h>
```

Comment:
I think with `#include "wasm-rt-impl.h"` (quotes instead of angle brackets) we can avoid having to add the `target_include_directories` line to CMakeLists.txt.